Radzen.Blazor/QueryableExtension.cs

```diff
@@ -583,18 +583,18 @@ private static string GetColumnODataFilter<T>(RadzenGridColumn<T> column, bool s
         /// </summary>
         /// <typeparam name="T"></typeparam>
         /// <param name="column">The column.</param>
-        /// <param name="second">if set to <c>true</c> [second].</param>
+        /// <param name="filterValue">The specific value to filter by</param>
+        /// <param name="columnFilterOperator">The operator used to compare to <paramref name="filterValue"/></param>
         /// <returns>System.String.</returns>
-        private static string GetColumnODataFilter<T>(RadzenDataGridColumn<T> column, bool second = false)
+        internal static string GetColumnODataFilter<T>(RadzenDataGridColumn<T> column, object filterValue, FilterOperator columnFilterOperator)
         {
             var property = column.GetFilterProperty().Replace('.', '/');
 
-            var columnFilterOperator = !second ? column.GetFilterOperator() : column.GetSecondFilterOperator();
             var odataFilterOperator = ODataFilterOperators[columnFilterOperator];
 
-            var value = IsEnumerable(column.FilterPropertyType) && column.FilterPropertyType != typeof(string) ? null :
-                    !second ? (string)Convert.ChangeType(column.GetFilterValue(), typeof(string)) :
-                        (string)Convert.ChangeType(column.GetSecondFilterValue(), typeof(string));
```

Comment:
Looks like the second filter value is missing in the new implementation?

Reply:
The responsibility for choosing which value to filter on has been moved to RadzenDataGridColumn.GetColumnODataFilter. This way, QueryableExtension can focus more on generating single OData expressions and less on the inner workings of the data grid column, improving maintainability. Maybe in the future there's a third filterable value, or even a list of acceptable values, for example.